test/hotspot/gtest/nmt/test_nmt_buffer_overflow_detection.cpp

```diff
@@ -35,8 +35,15 @@
 // come, just use this one.
 #define COMMON_NMT_HEAP_CORRUPTION_MESSAGE_PREFIX "NMT corruption"
 
+#ifdef ADDRESS_SANITIZER
+#define ASAN_NOT_ENABLED() false
+#else
+#define ASAN_NOT_ENABLED() true
+#endif
+
 #define DEFINE_TEST(test_function, expected_assertion_message)                            \
-  TEST_VM_FATAL_ERROR_MSG(NMT, test_function, ".*" expected_assertion_message ".*") {     \
+  TEST_VM_FATAL_ERROR_MSG_IF(ASAN_NOT_ENABLED(), NMT, test_function,                      \
+                             ".*" expected_assertion_message ".*") {                      \
```

Comment:
I'd prefer a simple ifndef ADDRESS_SANITIZER instead of xxx_IF. And a comment why this does not work with asan.